lib/elasticsearchHandler.js

```diff
@@ -60,18 +143,42 @@ ElasticsearchStore.prototype.initialise = function(resourceConfig) {
     self._db = client;
   }
   self.resourceConfig = resourceConfig;
-  if (!self._db) {
-    return elasticsearch.resourcesToInitialise.push(resourceConfig);
-  }
   self.ready = true;
 };
 
+ElasticsearchStore.prototype.populate = function(callback) {
+  var self = this;
+  self._db.indices.delete({ index: "jsonapi" }, function(err) {
+    if (err) console.log("Error dropping index?", err.message);
+
+    self._db.indices.create({ index: "jsonapi" }, function(err1) {
+      if (err1) console.log("Error creating index?", err1);
+
+      var mappingRequest = {
+        index: "jsonapi",
+        type: self.resourceConfig.resource,
+        body: self._generateMappingFor(self.resourceConfig)
+      };
+      self._db.indices.putMapping(mappingRequest, function(err2) {
```

Comment:
This triggered my `> 2` nested callbacks alarm... Could we possibly convert it to a `async.waterfall`?